akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -2185,6 +2184,8 @@ private[stream] object Collect {
 
       private def switchTo(flow: Flow[I, O, M], firstElement: I): M = {
 
+        var firstElementPushed = false
```

Comment:
I think it would be better to capture `firstElement` in an `OptionVal` `var` here and use the presence of that as the flag that it needs to be pushed, and then toggle it to `OptionVal.None` after emitting it. That'd avoid  keeping a reference to `firstElement` for the rest ot the life of the inner flow (hindering it from getting GC:d)

Reply:
Done :)